conans/client/tools/env.py

```diff
@@ -111,3 +113,45 @@ def remove_from_path(command):
 
     with environment_append({"PATH": curpath}):
         yield
+
+
+def env_diff(cmd, only_diff):
+    known_path_lists = ("include", "lib", "libpath", "path")
```

Comment:
Also probably want `LD_LIBRARY_PATH`, `DYLD_LIBRARY_PATH`, and `MANPATH` if you are going to hardcode path lists. I would probably avoid hardcoding path lists here, as there are a myriad of other ones as well (e.g. `PERL5LIB` and `PYTHONPATH`).

Reply:
that was always here, I've just extracted code from `vcvars_dict`, as it already has a well-tested implementation.

Reply:
for now, I think I'm not going to touch these variables unless it's absolutely required. I've no idea why things like include/lib need special handling here.

Reply:
Fundementally they are lists, not simple strings. They are needed for something like this in a conanfile:
```py
def build(self):
    env = tools.vcvars_dict(self.settings)
    env["INCLUDE"].append("C:\\path\\where\\more\\needed\\headers\\are\\located")
    with tools.environment_append(env):
       ...
```
If you don't specially treat them, this code wouldn't work because `env["INCLUDE"]` would just be a string.

I've written a good number of conanfiles that have had to extend the base compiler environment (either vcvars or compilervars), and treating environment variables that are `os.pathsep` separated lists as lists rather than strings dramatically simplifies the conanfile.

Reply:
good, now I understand the point
I've added more environment variables of such nature. feel free to comment if I missed something (I believe there is a ton of them).

Reply:
I'm not noticing any that I regularly use that are missing.